Added randomly generated marker placement

elad-haviv opened the pull request:
# Description

- I needed an option to generate random markers of specific types for my map.
- Added new functionality to the "Markers generation settings" window.
- The feature adds another column to the marker list with a button that enables the user to place a new randomly generated marker of that type on the map.
- 

![image](https://github.com/Azgaar/Fantasy-Map-Generator/assets/6059020/bb31771d-8971-43b4-8abb-ecb3df58801e)



<!-- Please include a summary of the change, motivation and context. It it's a but fix, add a reference in format #issue_number -->

# Type of change

<!-- Please put X into brackers of required option OR delete options that are not relevant -->

- [ ] Bug fix
- [X] New feature
- [ ] Refactoring / style
- [ ] Documentation update / chore
- [ ] Other (please describe)

# Versioning

<!-- Update the version if you want the PR to be merged fast. Currently it's a manual 3-steps process:
  * update version in `versioning.js` using semver principle. Just set the next patch (for fixes) or minor version (for new features)
  * for all changed files update hash (the part after `?`) in place where file is requested (usually it's `index.html`)
  * if the change can be really interesting for end-users, describe it inside the `showUpdateWindow()` function in `versioning.js` -->

- [x] Version is updated
- [x] Changed files hash is updated

Azgaar commented on versioning.js:
```diff
@@ -1,7 +1,7 @@
 
 // version and caching control
-const version = "1.89.36"; // generator version, update each time
+const version = "1.90.00"; // generator version, update each time
```
It's a minor change, I would prefer a 3rd parameter ("parch") increment, i.e. "1.89.37". We change the 2nd parameter ("minor version") when there are changes affecting old maps. So it's a bit different from a classical SemVer.

Azgaar commented:
Hello. It looks good! Should not take a lot of time to do the corrections and then I will test it additionally and merge.

Azgaar commented:
By the way, I am not sure it should be there in `Markets generation settings`. It's not a setting and not related to that menu. There is already a functionality to add a marker on click. But not it just creates a marker of the latest type. Can this button be reused to open a menu to add a marker of a selected type? I would be more semantically correct

elad-haviv commented:
> By the way, I am not sure it should be there in `Markets generation settings`. It's not a setting and not related to that menu. There is already a functionality to add a marker on click. But not it just creates a marker of the latest type. Can this button be reused to open a menu to add a marker of a selected type? I would be more semantically correct

How about adding a selection field in the Markers Overview window? 
When the user clicks on the standard '+' button, it could generate a random marker of the selected type. 
We can set the default value to 'empty,' retaining the button's current functionality.
What do you think?

Azgaar commented:
Yes, it should be fine in general. It looks a bit clumsy as there is no space for it. Also type selection toggles the add button state, it's fine, but weird if you select something again and button is getting unselected

elad-haviv commented:
> Yes, it should be fine in general. It looks a bit clumsy as there is no space for it. Also type selection toggles the add button state, it's fine, but weird if you select something again and button is getting unselected

I fixed it so type selection activates the add button state instead of toggling it.
also changed the selection menu to a button so it's less clumsy, is that okay?

Azgaar commented:
Yes, it looks good
